java/src/main/java/io/vertx/ext/eventbus/client/EventBusClientOptions.java

```diff
@@ -0,0 +1,409 @@
+package io.vertx.ext.eventbus.client;
+
+import io.vertx.ext.eventbus.client.options.ProxyOptions;
+import io.vertx.ext.eventbus.client.options.TcpTransportOptions;
+import io.vertx.ext.eventbus.client.options.TrustOptions;
+import io.vertx.ext.eventbus.client.options.WebSocketTransportOptions;
+
+/**
+ * @author <a href="mailto:pl@linux.com">Phil Lehmann</a>
+ *         <p>
+ *         Based on io.vertx.core.net.ClientOptionsBase and others by @author Tim Fox
+ */
+public class EventBusClientOptions {
+
+  /**
+   * The default value for host name = "localhost"
+   */
+  public static final String DEFAULT_HOST = "localhost";
+
+  /**
+   * The default value for port = 8080
+   */
+  public static final int DEFAULT_PORT = 8080;
+
+  /**
+   * The default value of connect timeout = 60000 ms
+   */
+  public static final int DEFAULT_CONNECT_TIMEOUT = 60000;
+
+  /**
+   * The default value of ping interval = 5000 ms
+   */
+  public static final int DEFAULT_PING_INTERVAL = 5000;
+
+  /**
+   * SSL enable by default = false
+   */
+  public static final boolean DEFAULT_SSL = false;
+
+  /**
+   * Default idle timeout = 0 ms (0 = disabled)
+   */
+  public static final int DEFAULT_IDLE_TIMEOUT = 0;
+
+  /**
+   * Default value of whether hostname verification (for SSL/TLS) is enabled = true
+   */
+  public static final boolean DEFAULT_VERIFY_HOST = true;
+
+  /**
+   * The default value of whether all servers (SSL/TLS) should be trusted = false
+   */
+  public static final boolean DEFAULT_TRUST_ALL = false;
+
+  /**
+   * The default value of whether auto reconnects are enabled, even if the client does not try to send a message = true
+   */
+  public static final boolean DEFAULT_AUTO_RECONNECT = true;
+
+  /**
+   * The default value of the pause between reconnect tries = 3000 ms
+   */
+  public static final int DEFAULT_AUTO_RECONNECT_INTERVAL = 3000;
+
+  /**
+   * The default value of the maximum number of auto reconnect tries = 0 (0 = no limit)
+   */
+  public static final int DEFAULT_MAX_AUTO_RECONNECT_TRIES = 0;
+
+  private TcpTransportOptions tcpTransportOptions;
```

Comment:
we should have one field transportOptions and not two that is TransportOptions

Reply:
Thanks for your review! Your comments seem to be straighforward, I'll get to it.